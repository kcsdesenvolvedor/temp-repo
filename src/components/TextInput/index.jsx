import { React } from 'react';
import P from 'prop-types';
import './style.css';

export const TextInput = ({ searchValue, handleCharge }) => {
  return (
    <input className="input" type="search" onChange={handleCharge} value={searchValue} placeholder="input for search" />
  );
};

TextInput.propTypes = {
  searchValue: P.string.isRequired,
  handleCharge: P.func.isRequired,
};
