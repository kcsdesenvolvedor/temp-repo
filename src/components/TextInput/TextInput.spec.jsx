import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { TextInput } from '../TextInput/index';
import { React } from 'react';

describe('<TextInput />', () => {
  it('should have a value of searchValue', () => {
    const fn = jest.fn();
    render(<TextInput handleCharge={fn} searchValue={'testando'} />);

    const input = screen.getByPlaceholderText(/input for search/i);
    expect(input).toBeInTheDocument();

    expect(input.value).toBe('testando');
  });

  it('should call handleChange function on each key pressed', () => {
    const fn = jest.fn();
    render(<TextInput handleCharge={fn} searchValue="um valor qualquer" />);

    const input = screen.getByPlaceholderText(/input for search/i);

    const value = 'o valor';

    userEvent.type(input, value);

    expect(input.value).toBe('um valor qualquer');
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it('should match snapshot', () => {
    const fn = jest.fn();
    const { container } = render(<TextInput handleCharge={fn} searchValue="" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
