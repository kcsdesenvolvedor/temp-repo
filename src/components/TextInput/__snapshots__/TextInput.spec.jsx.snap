// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match snapshot 1`] = `
<input
  class="input"
  placeholder="input for search"
  type="search"
  value=""
/>
`;
